lab4/provided/Makefile

```diff
@@ -1,3 +1,33 @@
+CXX := g++
+CXXFLAGS := -std=gnu++11 -g -DDEBUG
 
-cache_simulator: main.cc cache.cc direct_mapped.cc memory.cc ticked_object.cc processor.cc tag_array.cc non_blocking.cc set_assoc.cc
-	g++ -std=gnu++11 main.cc cache.cc direct_mapped.cc memory.cc ticked_object.cc processor.cc tag_array.cc non_blocking.cc set_assoc.cc -o cache_simulator -g -DDEBUG
+all: cache_simulator
+
+objs := \
+	cache.o \
+	direct_mapped.o \
+	main.o \
+	memory.o \
+	non_blocking.o \
+	processor.o \
+	set_assoc.o \
+	tag_array.o \
+	ticked_object.o
+
+DEPFLAGS = -MMD -MF $(@:.o=.d)
+deps := $(patsubst %.o,%.d,$(objs))
+-include $(deps)
```

Comment:
This is more makefile magic than I know...

Reply:
This looks okay actually, the lines above just quickly set some dependency file references. Honestly my only bad comment is that the -g and debug flags should be kept separate from the main flags, and adjustable by Makefile argument.

Reply:
+1, good idea.

Reply:
Done @animalvgamer.